incubator/hnc/pkg/testutils/testutils.go

```diff
@@ -229,7 +232,49 @@ func RunCommand(cmdln ...string) (string, error) {
 	stdout, err := cmd.CombinedOutput()
 	return string(stdout), err
 }
+// CreateNamespace creates the specified namespace with canned testing labels making it easier
+// to look up and delete later.
+func CreateNamespace( ns string)   {
+	CreateNamespaceWithLabels(ns,"", testingNamespaceLabels)
+}
+// CreateSubnamespace creates the specified namespace in the parent namespace with canned testing labels making it easier
+// to look up and delete later.
+func CreateSubnamespace( ns string,parent string)   {
+	CreateNamespaceWithLabels(ns,parent, testingNamespaceLabels)
+}
+
+// CreateNamespaceWithLabels creates a namespace or a subnamespace (if a parent is specified) and marks the namespace
+// with all the labels specified
+func CreateNamespaceWithLabels( ns string, parent string, labels map[string]string)  {
+	if parent == ""{
+		MustRun("kubectl create ns", ns)
+	}else{
+		MustRun("kubectl hns create", ns, "-n", parent)
+	}
+	for k,v:= range labels{
+		label:= fmt.Sprintf("%s=%s",k,v)
+		MustRun("kubectl label --overwrite ns", ns,label)
+	}
+
+}
+// CleanupTestNamespaces finds the list of namespaces labeled as test namespaces and deleg
+// CleanupNamespaces function. These two can potentially be combined if there is no specific need
+// for a function to cleanup a specific namespace like CleanupNamespaces
+func CleanupTestNamespaces(){
+	labels:= make([]string, 0)
+	for k,v:= range testingNamespaceLabels{
+		labels=append(labels,fmt.Sprintf("%s=%s",k,v))
+	}
+	labelQuery := strings.Join(labels,",")
+	out,err:=RunCommand("kubectl get ns -o custom-columns=:.metadata.name --no-headers=true","-l",labelQuery)
+	if err !=nil{
+		GinkgoT().Log("Unable to get namespaces (failed): ", err)
+	}else{
+		nses:= strings.Split(out,"\n")
+		CleanupNamespaces(nses...)
```

Comment:
This should now be the only place that `CleanupNamespaces` is called, so either make it private (e.g. start with a lowercase letter - `cleanupNamespaces`) or just fold everything from that function into this one (assuming you've gotten rid of the indent).

Reply:
Yes, again the reason to leave it public was have future test case wanting to delete a specific namespace. If not envisioned i will make it private.